javascript/test/basic-tests.js

```diff
@@ -8,103 +9,137 @@ const { stdin } = require('mock-stdin');
+describe('Basic Tests', function () {
+
+    describe('Runs correct version of node. Tag: Assignment1, Assignment2, Assignment3, Assignment4, Assignment5', function () {
+        it('Version of node must be v10.23.0', function () {
             assert.strictEqual(process.version, 'v10.23.0');
         });
     });
 
     // https://stackoverflow.com/questions/21587122/mocha-chai-expect-to-throw-not-catching-thrown-errors
     // assert.throws( FunctionThatShouldThrow_AssertionError, assert.AssertionError )
-    describe('Runs with non-null exit codes. Tag: Assignment1, Assignment2, Assignment3, Assignment4, Assignment5', function() {
+    describe('Runs with non-null exit codes. Tag: Assignment1, Assignment2, Assignment3, Assignment4, Assignment5', function () {
+
+        var tempDir;
+        var minesweeperProcess;
 
-        beforeEach("Ensure no simple.cfg before execution.", function(){
-            if( fs.existsSync("simple.cfg")){
-                fs.unlinkSync("simple.cfg");
-            }
+        beforeEach("Create temporary folder for file creation", function () {
+            tempDir = fs.mkdtempSync(os.tmpdir() + path.sep);
         });
 
-        afterEach("Ensure no simple.cfg after execution", function(){
-            if( fs.existsSync("simple.cfg")){
-                fs.unlinkSync("simple.cfg");
-            }
+        afterEach("Remove temporary folder for file creation", function () {
+            // storing the global variable into a local one for each test might be a bit of a hack,
+            // but because of the asynchronous behavior there is not really another chance (to the best of my knowledge)
+            // unless abstaining from the usage of before/afterEach.
+            var tempDirLocal = tempDir;
+            minesweeperProcess.on('close', () => {
+                deleteFolderWithFiles(tempDirLocal);
+            });
         });
 
-        it('Minesweeper should exit with code 1 if input file is not provided', function() {
-            var exitCode = minesweeper.main()
-            assert.strictEqual(exitCode, 1, "Wrong exit code for missing board configuration file")
+        it('Minesweeper should exit with code 1 if input file is not provided', function () {
+            minesweeperProcess = spawn('node', [minesweeperHome + "minesweeper.js"]);
+
+            minesweeperProcess.on('close', (exitCode) => {
+                assert.strictEqual(exitCode, 1, "Wrong exit code for missing board configuration file")
+            });
         });
 
-        it('Minesweeper should exit with code 1 if input file does not exist', function() {
-            var exitCode = minesweeper.main("simple.cfg")
-            assert.strictEqual(exitCode, 1, "Wrong exit code for not existent board configuration file")
+        it('Minesweeper should exit with code 1 if input file does not exist', function () {
+            minesweeperProcess = spawn('node', [minesweeperHome + "minesweeper.js", "simple.cfg"]);
```

Comment:
Shouldn't this be something like `tempDir + simple.cfg`? The point is: if someone has an existing `simple.cfg` file, the test might fail for the wrong reason. Instead, by "pretending" the file is in the tempDir we should be sure that indeed there will be no file as the test expects.

Question: `tempDir` is defined as `os.tmpdir() + path.sep` on line 44, doesn't this implies that there's a tralining "/" already?

Reply:
It should be indeed tempDir + simple.cfg, yes. Otherwise exactly the issue mentioned might happen.

tempDir is not exactly defined as os.tmpdir() + path.sep, but as a subfolder of it (using fs.mkdtempSync). So the trailing "/" is necessary because mkdtempSync returns the path without this sign. However, it might be a bit error-prone that every time a config file is created, the separator has to be written. So I append the separator immediately in the beforeEach function (cf. recent commit).